src/lib.rs

```diff
@@ -334,6 +335,67 @@ extern crate indexmap;
 extern crate itoa;
 extern crate ryu;
 
+////////////////////////////////////////////////////////////////////////////////
+
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
+/// A facade around all the types we need from the `std`, `core`, and `alloc`
+/// crates. This avoids elaborate import wrangling having to happen in every
+/// module.
+mod lib {
+    mod core {
+        #[cfg(not(feature = "std"))]
+        pub use core::*;
+        #[cfg(feature = "std")]
+        pub use std::*;
+    }
+
+    pub use self::core::{char, str};
+    pub use self::core::{cmp, mem, num, slice};
+
+    pub use self::core::{borrow, iter, ops};
+
+    pub use self::core::cell::{Cell, RefCell};
+    pub use self::core::clone::{self, Clone};
+    pub use self::core::convert::{self, From, Into};
+    pub use self::core::default::{self, Default};
+    pub use self::core::fmt::{self, Debug, Display};
+    pub use self::core::hash::{self, Hash};
+    pub use self::core::marker::{self, PhantomData};
+    pub use self::core::result::{self, Result};
+
+    #[cfg(all(feature = "alloc", not(feature = "std")))]
```

Comment:
This can be simplified to `not(feature = "std")` because we know alloc must be enabled in that case. Same thing below.

In general `feature = "alloc"` should not need to appear anywhere.

Reply:
I did that to introduce distinction between `std`, `alloc` and no feature at all, but maybe it'd be good only introduce two: default `std` and no-`std` (which implies `alloc` and is supported from 1.36 onwards)...

I'll adapt the code accordingly, thanks!